plotly/tools.py

```diff
@@ -1570,6 +1578,43 @@ def _validate_streamline(x, y):
                                              "evenly spaced array")
 
     @staticmethod
+    def _validate_annotated_heatmap(z, annotation_text):
+        """
+        Annotated heatmap specific validations
+
+        Check that if a text matrix is supplied, it is the same
+        dimmensions as the z matrix.
+
+        See FigureFactory.create_annotated_heatmap() for params
+
+        :raises: (PlotlyError) If z and text matrices are not the same
+            dimmensions.
+        """
+        if annotation_text is not None and isinstance(annotation_text, list):
+            FigureFactory._validate_equal_length(z, annotation_text)
+            for lst in range(len(z)):
+                if len(z[lst]) != len(annotation_text[lst]):
+                    raise exceptions.PlotlyError("z and text should have the "
+                                                 "same dimmensions")
+
+    @staticmethod
+    def _validate_table(table_text, fontcolor):
+        """
+        Table specific validations
+
+        Check that the fontcolor is supplied correctly (1, 3, or len(text)
+            colors).
+
+        :raises: (PlotlyError) If fontcolor is supplied incorretly.
+
+        See FigureFactory.create_table() for params
+        """
+        fontcolor_len_options = [1, 3, len(table_text)]
+        if len(fontcolor) not in fontcolor_len_options:
+            raise exceptions.PlotlyError("Oops, fontcolor should be a list of "
+                                         "length 1, 3 or len(text)")
```

Comment:
:cow2: Does `fontcolor` line up with a plotlyjs attribute? Otherwise, the fact that it's an array is a little misleading. Maybe `fontcolors` or `font_colors`?